src/server/master_server/chunk_server_manager.h

```diff
@@ -0,0 +1,113 @@
+#ifndef GFS_SERVER_MASTER_SERVER_CHUNK_SERVER_MANAGER_H_
+#define GFS_SERVER_MASTER_SERVER_CHUNK_SERVER_MANAGER_H_
+
+#include <memory>
+#include <string>
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/hash/hash.h"
+
+namespace gfs {
+
+// The unique identifier for a chunk server
+class ChunkServerKey {
+  // To make ChunkServerKey hashable, for use in absl hash map.
+  template <typename H>
+  friend H AbslHashValue(H hash_state, const ChunkServerKey& key);
+
+ public:
+  ChunkServerKey(const std::string& host_name, const std::string& port)
```

Comment:
Using the protos in our core logic isn't a good design decision. Shouldn't be taking such hard dependency on external framework. What if a better framework is released in the future and you want to use that? It becomes an expensive effort since you have dirtied your entire code base with proto stuffs. **I wouldn't do this in a different project (production system) but given the scope of this work, I have changed it.** My initial plan was to have something that does the conversion. In this case, if a new framework is released by company X that you would like to use, you just need to add a converter for this new framework and your core logic doesn't know about this change. Think this is cleaner and allows a project to easily evolve and not locked to some tech.

Reply:
Fair point. Yeah, if we don't want dependency on protocol buffers and it's a passive data storage, I will also use `struct` over `class` unless there are fancier operations we need. However, since we are using gRPC, having either `struct` or `class` will mean constant conversions between gRPC's protobuf and your class/struct, which can be expensive and repetitive -- a reason why I prefer protobuf here since it's already sent by the gRPC request